ctk/common/src/main/kotlin/org/codice/compliance/verification/binding/BindingVerifier.kt

```diff
@@ -53,8 +56,16 @@ abstract class BindingVerifier {
                                 "\n$IDP_ERROR_RESPONSE_REMINDER_MESSAGE")
             }
         }
-    }
 
-    abstract fun verifyError()
-    abstract fun verify()
+        fun getBindingVerifier(response: Response): BindingVerifier {
+            return when (response.determineBinding()) {
+                SamlProtocol.Binding.HTTP_REDIRECT -> RedirectBindingVerifier(response)
+                SamlProtocol.Binding.HTTP_POST -> PostBindingVerifier(response)
+                else -> throw UnsupportedOperationException("Binding is not currently supported.")
+            }
+        }
+    }
+    var isRelayStateGiven: Boolean = false
```

Comment:
❓ Should this be a field that is passed or a field that is determined from parsing the SAMLRequestDom?

Reply:
Sidenote:
If we can pull a lot of the test-specific information from the `SAMLRequestDom` instead of values passed in by the tests (e.g. figuring out if we passed the IdP a RelayState by parsing the `SAMLRequestDom` rather than by sending a flag called `isRelayStateGiven` into the verification), then we might be able to pull a lot of the verifications we call in each test class into a private method.

Reply:
It would have to be parsed from the HTTP request, not the SAML Request DOM.

I initially had that, but it felt odd to me. The tests do all this work to encode the relay state and put it in the Redirect or Post request, and then give that HTTP request to the BindingVerifier to parse out what the test **just** put there. Why not just give it directly to the BindingVerifier. I think if, down the road, the BindingVerifier needs something else from the HTTP request we should put it and the relay state into a data class and pass it down to the BindingVerifier.